contrib/zpages/src/main/java/io/opencensus/contrib/zpages/TraceConfigzZPageHandler.java

```diff
@@ -88,80 +98,86 @@ public void emitHtml(Map<String, String> queryMap, OutputStream outputStream) {
+    } finally {
+      // TODO(bdrutu): Maybe display to the page if an exception happened.
+      // Display the page in any case.
+      TraceParams currentParams = traceConfig.getActiveTraceParams();
+      out.printf(
+          TRACECONFIGZ_FORM_BODY,
+          CHANGE,
+          PERMANENT_CHANGE,
+          QUERY_COMPONENT_SAMPLING_PROBABILITY,
+          "0.0001", // TODO(bdrutu): Get this from the default sampler (if possible).
+          QUERY_COMPONENT_MAX_NUMBER_OF_ATTRIBUTES,
+          TraceParams.DEFAULT.getMaxNumberOfAttributes(),
+          QUERY_COMPONENT_MAX_NUMBER_OF_ANNOTATIONS,
+          TraceParams.DEFAULT.getMaxNumberOfAnnotations(),
+          QUERY_COMPONENT_MAX_NUMBER_OF_NETWORK_EVENTS,
+          TraceParams.DEFAULT.getMaxNumberOfNetworkEvents(),
+          QUERY_COMPONENT_MAX_NUMBER_OF_LINKS,
+          TraceParams.DEFAULT.getMaxNumberOfLinks());
+      out.write("<br>\n");
+      out.printf(RESTORE_DEFAULT_FORM_BODY, CHANGE, RESTORE_DEFAULT_CHANGE);
+      out.write("<br>\n");
+      emitTraceParamsTable(currentParams, out);
+      out.write("</body>\n");
+      out.write("</html>\n");
+      out.close();
     }
-    TraceParams currentParams = traceConfig.getActiveTraceParams();
-    out.printf(
-        TRACECONFIGZ_FORM_BODY,
-        PERMANENT_CHANGE,
-        PERMANENT_CHANGE,
-        QUERY_COMPONENT_SAMPLING_PROBABILITY,
-        0.0001, // TODO(bdrutu): Get this from the current sampler (if possible).
-        CHANGE_ON_INPUT_PERMANENT,
-        QUERY_COMPONENT_MAX_NUMBER_OF_ATTRIBUTES,
-        currentParams.getMaxNumberOfAttributes(),
-        CHANGE_ON_INPUT_PERMANENT,
-        QUERY_COMPONENT_MAX_NUMBER_OF_ANNOTATIONS,
-        currentParams.getMaxNumberOfAnnotations(),
-        CHANGE_ON_INPUT_PERMANENT,
-        QUERY_COMPONENT_MAX_NUMBER_OF_NETWORK_EVENTS,
-        currentParams.getMaxNumberOfNetworkEvents(),
-        CHANGE_ON_INPUT_PERMANENT,
-        QUERY_COMPONENT_MAX_NUMBER_OF_LINKS,
-        currentParams.getMaxNumberOfLinks(),
-        CHANGE_ON_INPUT_PERMANENT);
-    emitTraceParamsTable(currentParams, out);
-    out.write("</body>\n");
-    out.write("</html>\n");
-    out.close();
   }
 
   // If this is a supported change (currently only permanent changes are supported) apply it.
   private void maybeApplyChanges(Map<String, String> queryMap) {
-    if (queryMap.containsKey(CHANGE) && queryMap.get(CHANGE).equals(PERMANENT_CHANGE)) {
+    String changeStr = queryMap.get(CHANGE);
+    if (PERMANENT_CHANGE.equals(changeStr)) {
       TraceParams.Builder traceParamsBuilder = traceConfig.getActiveTraceParams().toBuilder();
-      if (queryMap.containsKey(QUERY_COMPONENT_SAMPLING_PROBABILITY)) {
-        double samplingProbability =
-            Double.parseDouble(queryMap.get(QUERY_COMPONENT_SAMPLING_PROBABILITY));
+      String samplingProbabilityStr = queryMap.get(QUERY_COMPONENT_SAMPLING_PROBABILITY);
+      if (samplingProbabilityStr != null && !samplingProbabilityStr.isEmpty()) {
```

Comment:
Suggestion: You could use `com.google.common.base.Strings.isNullOrEmpty` here and below.

Reply:
Done.